extensions/jms/src/main/java/org/mule/extensions/jms/api/source/JmsListener.java

```diff
@@ -172,23 +178,35 @@ public void onStop() {
     if (LOGGER.isDebugEnabled()) {
       LOGGER.debug("Stopping JMSSubscriber source");
     }
+
+    jmsLock.unlock();
+
+    try {
+      if (ackMode.equals(AUTO) || ackMode.equals(DUPS_OK) || ackMode.equals(MANUAL)) {
+        session.get().recover();
+      }
+    } catch (JMSException e) {
+      throw new JmsExtensionException(e, "A problem occurred recovering the session before returning the session to the pool");
+    }
   }
 
   @OnSuccess
   public void onSuccess(@Optional @NullSafe JmsListenerResponseBuilder response,
                         SourceCallbackContext callbackContext) {
-
-    ackOriginalMessage(callbackContext);
-
+    jmsLock.unlock();
```

Comment:
here we completed the processing of the flow, but I don't get where is the AUTO ack performed. Are you doing that behind the curtains?